src/App.tsx

```diff
@@ -1,61 +1,159 @@
+  const [errorUser, setErrorUser] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+    setErrorTitle(false);
+  };
+
+  const resetFields = () => {
+    setTitle('');
+
+    setErrorUser(false);
+    setSelectedUserId(0);
+  };
+
+  const handleUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setErrorUser(false);
+    setSelectedUserId(+value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserId) {
+      setErrorUser(true);
+    }
+
+    setTitle(title.replace(/[^A-Za-z\s\d\u0400-\u04FF]/g, ''));
+
+    if (title.trim() && selectedUserId) {
+      const newTodo: Todo = {
+        id: Math.max(...todos.map((todo) => todo.id)) + 1,
+        userId: selectedUserId,
+        title,
+        completed: false,
+        user: getUser(selectedUserId),
+      };
+
+      setTodos((prevTodo => [...prevTodo, newTodo]));
+
+      resetFields();
+    }
+  };
+
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+    <div className="containerForApp">
+      <div className="App">
+        <h1 className="title">
+          Add todo form
+        </h1>
+
+        <form
+          action="/api/users"
```

Comment:
You can remove the 'action' and 'method' attributes from the form element, as they are not needed when handling form submission with React.
Overall, good job with adding them for backwards compatibility

Reply:
Fix Vadim's comment